File: C++_implementation/GraphReader.cpp
#include "GraphReader.h"
#include <iostream>
using namespace std;

GraphReader::GraphReader(const string& filePath) {
    this->filePath = filePath; 
}


 NetworKit::Graph GraphReader::readEdgesFormatGraph() const {
    
    char separator = ' ';        
    int firstNode = 0;           
    string commentPrefix = "#"; 
    bool continuous = true;      
    bool directed = false;      

    
    NetworKit::EdgeListReader reader(separator, firstNode, commentPrefix, continuous, directed);

    
    NetworKit::Graph graph = reader.read(filePath);

    return graph;
}

File: python_implementation/GraphReader.py
from networkit import graphio
import networkit as nk
class GraphReader:
    def __init__(self, file_path):
        self.file_path = file_path

    def read_edges_format_graph(self):
        separator = ' '          # Separator for the edge list file
        first_node = 0           # Whether node numbering starts at 0
        comment_prefix = "#"     # Prefix for comment lines
        continuous = True        # Whether nodes are continuously numbered
        directed = False         # Whether the graph is directed

        reader = graphio.EdgeListReader(" ", 0, "#", True, False)
    	G = reader.read(self.file_path)
        
        return G